src/Data/MessagePack/Generic.hs

```diff
@@ -18,6 +18,9 @@ import           GHC.Generics
 import           Data.MessagePack.Class
 import           Data.MessagePack.Object (Object (..))
 
+instance GMessagePack V1 where
+  gToObject = undefined
+  gFromObject = fail "invalid encoding for void type"
```

Comment:
I see - there might be a test that checks this. "can't instantiate void type" sounds fine. Are there tests that fail if you do this?